tdp_server/services/deployment_plan.py

```diff
@@ -87,6 +87,7 @@ async def from_reconfigure_request(
         latest_successes = db.execute(
             get_latest_success_service_component_version_query()
         ).all()
+        db.close()
```

Comment:
This is not the right way to fix that issue. 
These two paths do not use the session in other threads, and the session is closed automatically on leaving request execution.

You need to follow path execution and see where a session has been instantiated on the main thread, and then used on a worker thread.